mm2src/coins/hd_wallet_storage/mod.rs

```diff
@@ -0,0 +1,543 @@
+#[cfg(target_arch = "wasm32")] mod wasm_storage;
+
+#[cfg(any(test, target_arch = "wasm32"))] mod mock_storage;
+#[cfg(any(test, target_arch = "wasm32"))]
+pub use mock_storage::HDWalletMockStorage;
+
+cfg_wasm32! {
+    use wasm_storage::HDWalletIndexedDbStorage as HDWalletStorageInstance;
+
+    pub use wasm_storage::{HDWalletDb, HDWalletDbLocked};
+}
+
+cfg_native! {
+    use sqlite_storage::HDWalletSqliteStorage as HDWalletStorageInstance;
+}
+
+pub type HDWalletStorageResult<T> = MmResult<T, HDWalletStorageError>;
+type HDWalletStorageBoxed = Box<dyn HDWalletStorageInternalOps + Send + Sync>;
+
+#[derive(Debug, Display)]
+pub enum HDWalletStorageError {
+    #[display(fmt = "HD wallet not allowed")]
+    HDWalletUnavailable,
+    #[display(fmt = "HD account '{:?}':{} not found", wallet_id, account_id)]
+    HDAccountNotFound { wallet_id: HDWalletId, account_id: u32 },
+    #[display(fmt = "Error saving the a swap: {}", _0)]
+    ErrorSaving(String),
+    #[display(fmt = "Error loading a swap: {}", _0)]
+    ErrorLoading(String),
+    #[display(fmt = "Error deserializing a swap: {}", _0)]
+    ErrorDeserializing(String),
+    #[display(fmt = "Error serializing a swap: {}", _0)]
+    ErrorSerializing(String),
+    #[display(fmt = "Internal error: {}", _0)]
+    Internal(String),
+}
+
+impl From<CryptoInitError> for HDWalletStorageError {
+    fn from(e: CryptoInitError) -> Self { HDWalletStorageError::Internal(e.to_string()) }
+}
+
+impl HDWalletStorageError {
+    pub fn is_deserializing_err(&self) -> bool { matches!(self, HDWalletStorageError::ErrorDeserializing(_)) }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct HDWalletId {
+    coin: String,
+    /// RIPEMD160(SHA256(x)) where x is a pubkey with which mm2 is launched.
+    /// It's expected to be equal to [`MmCtx::rmd160`].
+    /// This property allows us to store DB items that are unique to each user (passphrase).
+    mm2_rmd160: String,
+    /// RIPEMD160(SHA256(x)) where x is a pubkey extracted from a Hardware Wallet device or passphrase.
+    /// This property allows us to store DB items that are unique to each Hardware Wallet device.
+    /// Please note it can be equal to [`HDWalletId::mm2_rmd160`] if mm2 is launched with a HD private key derived from a passphrase.
+    hd_wallet_rmd160: String,
+}
+
+impl HDWalletId {
+    pub fn new(coin: String, mm2_rmd160: &H160, hd_wallet_rmd160: &H160) -> HDWalletId {
+        HDWalletId {
+            coin,
+            mm2_rmd160: display_rmd160(mm2_rmd160),
+            hd_wallet_rmd160: display_rmd160(hd_wallet_rmd160),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct HDAccountStorageItem {
+    pub account_id: u32,
+    pub account_xpub: XPub,
+    /// The number of addresses that we know have been used by the user.
+    pub external_addresses_number: u32,
+    pub internal_addresses_number: u32,
+}
+
+#[async_trait]
+#[cfg_attr(any(test, target_arch = "wasm32"), mockable)]
```

Comment:
`mockable` generates additional highly unsafe code that I prefer to avoid in the production environment. Could you please check what it actually generates and try to remove the `target_arch = "wasm32"` part?

Reply:
Fortunately, `HDWalletMockStorage` is not used within WASM tests.
Done

Reply:
@sergeyboyko0791 As I can see, this specific line didn't change 🙂

Reply:
Oh sorry, I've missed to remove the `mockable` attribute  for the `HDWalletStorageInternalOps` trait...
Done